### PR TITLE
Analysis #69 DC vacant lot analysis

### PR DESCRIPTION
**Pull request by maggiesu0725**

### CODE REVIEW

**Comment by maggiesu0725** [comment]
@lhendey I'm still working on reprojecting the parcel base geography for spatial joining instead of using the blacknight lat and lon. But since DC blacknight data zoning information is outdated, the program is a bit different than Sarah's example. I thought it would be helpful for you to just take a look at the structure and procedure so I can adjust accordingly while figuring out some programming issues. Thanks!

**Comment by maggiesu0725** [comment]
@sstrochak will you have time to take a quick look at the DC version vacant lot?

**Comment by sstrochak** [comment]
@maggiesu0725 yeah, is it ready now?

**Comment by maggiesu0725** [comment]
I haven't solved the projection issue but just want to get some general feedback on quality checks. DC blacknight zoning is outdated so the part where you compared the blacknight and shapefile zoning is not applicable in this case, I also commented in the program where the original methodology is not applicable. If you could take a look at those. Thanks! @sstrochak

**Comment by sstrochak** [comment]
@maggiesu0725 I can get to it either tomorrow morning or Wednesday morning, does that work?

**Comment by maggiesu0725** [comment]
Sure take your time! or if you prefer I walk you through the code sometime tomorrow. @sstrochak

**Comment by maggiesu0725** [comment]
So I've subdivided the lots with #dwelling restrictions. Could you do a final review? @sstrochak @lhendey

**Comment by maggiesu0725** [comment]
@lhendey I made all the changes and added the number of observation I got in the comment so it will be easier for you to compare,

**Comment by lhendey** [comment]
@maggiesu0725 - my html file after knitting still has different #s. if you knit the markdown to html do you get the same result as me?

File: Prog/vacant-lots-DC.Rmd [hunk 1]
```diff
@@ -0,0 +1,886 @@
...
+ggplot(zoningdata1test, aes(gap ))+
+  geom_histogram(binwidth = 500)
+
+
+```
+
+
+## Calculate missing units
+
+There are several variables we will use to calculate how many units we can put on each vacant lot.
```
Comment (lhendey) [inline Prog/vacant-lots-DC.Rmd]: edit this comment to reflect dc calculations

**Review by lhendey — CHANGES_REQUESTED** [review]
@maggiesu0725 see comments. 

cc @sstrochak .

**Comment by maggiesu0725** [comment]
@lhendey  ready for review

File: Prog/vacant-lots-DC.Rmd [hunk 2]
```diff
@@ -0,0 +1,886 @@
...
+```{r antijoinagain}
+
+anti_join(vacantcleaned, zoningcode_factor, by = c("zoning_code")) %>% 
+  count(zoning_code) %>% 
+  knitr::kable()
+
+```
+
+There are 2228 R1s that are not joined which is a lot, in our spreadsheet we have R1A and R1B, pull in lotsize and see which ones we should assign the R1 to. R1A has lot size of 7500, R1B has lot size of 5000. Use as threshold.
+
```
[the remark this answers is not in the document]
Comment (lhendey) [reply Prog/vacant-lots-DC.Rmd]: @maggiesu0725 is this issue resolved?

**Comment by lhendey** [comment]
@maggiesu0725 - i corrected the living area variable in the units_far calc and it broke the scatter plot function...

is there anyway to save the result of the get_unit_summary - i'd like to look at a few of the obs  where units_Far is getting us a lot of units to figure out whether the calc is wrong.

File: Prog/vacant-lots-DC.Rmd [hunk 3]
```diff
@@ -0,0 +1,886 @@
...
+  dataset %>% 
+    mutate(parameters = paste0("CE: ", construction_efficiency * 100,
+                               "% , Unit size: ", average_unit_size,
+                               " sf"),
+           buildable_area = lotsize_sf * lot_coverage_max,
+           living_area = buildable_area * construction_efficiency,
+           units_per_floor = floor(living_area / average_unit_size),
+           floors = floor(height_ft / 10),
+           units_height = units_per_floor * floors,
+           units_FAR= (living_area*floor_area_ratio_max)/average_unit_size,
```
[the remark this answers is not in the document]
Comment (pspauster) [reply Prog/vacant-lots-DC.Rmd]: @lhendey do you know how we would adapt this calculation if there is no lot coverage specified?

**Comment by maggiesu0725** [comment]
I added the dataset **calculateunit_data**  which should be what you are looking for. It is in the calculation section. fixed on function and reknitted the document, should not be breaking now. @lhendey

**Comment by maggiesu0725** [comment]
also exported the dataset to CSV: L:/Libraries/RegHsg/Prog/Tables/DC-vacant-lots-analysis-basedata.csv @lhendey

**Comment by lhendey** [comment]
@maggiesu0725 thanks that's helpful. can you add the ssl to output?

Also - can you add units_far should be NA if floor_area_ratio_max is NA? (the zero's are trumping the units_height right now.

**Comment by maggiesu0725** [comment]
thanks, changes made. @lhendey

**Comment by maggiesu0725** [comment]
@lhendey I just knitted the document, if you want to take a look.

**Review by lhendey — COMMENTED** [review]
@maggiesu0725 

line 571 shouldn't we set jurtotal instead of jur2 to create jurcleaned?  they have different numbers of obs. 

line 621 jurzoningdata1 has less obs than jurtotal and jurcleaned. what is getting lost? can you add a comment?

**Comment by lhendey** [comment]
@maggiesu0725 just pushed a few minor changes.

**Comment by maggiesu0725** [comment]
@lhendey  I added these comments int eh program as well.
line 571 shouldn't we set jurtotal instead of jur2 to create jurcleaned? they have different numbers of obs.
No we are using jur2 because jur2 has only the observations that have either parcelbase geo or blacknight coordinates and joined with the zoning shapefile to get the zoning information. 

line 621 jurzoningdata1 has less obs than jurtotal and jurcleaned. what is getting lost? can you add a comment?
I'm filtering out observations with zoning code that are not in the DC spreadsheet, because they are not going to be calculated in the next step. I might need to take the filter out.

**Comment by lhendey** [comment]
@maggiesu0725 -FYI i'm working on the vacant-dc file

**Comment by sstrochak** [comment]
To reclassify the addresses:
```
#create jur4 from blackknight address
jur5 <- jur4 %>% 
  filter(category_detail!="Garage") %>%
  classify_parceladdresses() %>% 
  mutate(ID = row_number()) %>% 
  mutate(address_type = ifelse(propaddress %in% c("address", "address", "address"),
                                                 "single",
                                                  address_type))
```

**Comment by maggiesu0725** [comment]
> To reclassify the addresses:
> 
> ```
> #create jur4 from blackknight address
> jur5 <- jur4 %>% 
>   filter(category_detail!="Garage") %>%
>   classify_parceladdresses() %>% 
>   mutate(ID = row_number()) %>% 
>   mutate(address_type = ifelse(propaddress %in% c("address", "address", "address"),
>                                                  "single",
>                                                   address_type))
> ```

@sstrochak what's the context of this? Is this something I need to update in the program? @lhendey

**Comment by lhendey** [comment]
@maggiesu0725 - this is something i'm working on changing in the clean-dc file to get rid of lots that were incorrectly collapsing  and creating those huge # of units. cc @sstrochak

**Comment by maggiesu0725** [comment]
@lhendey ready for review.

**Review by lhendey — APPROVED** [review]

**Pull request merged**